lib/src/session.rs

```diff
@@ -252,6 +252,19 @@ impl Session {
             .ok_or(HandleError::InvalidBodyHandle(handle))
     }
 
+    /// Drop a [`Body`][crate::body::Body] from the [`Session`], given its [`BodyHandle`][crate::wiggle_abi::types::BodyHandle].
+    ///
+    /// Returns a [`HandleError`][crate::error::HandleError] if the handle is not associated with a body in the session.
+    pub fn drop_body(&mut self, handle: BodyHandle) -> Result<(), HandleError> {
+        self.bodies
+            .get_mut(handle)
+            .and_then(Option::take)
+            // Unlike take_body we don't want to call into_body
+            // as this will cause an InvalidBodyHandle error
```

Comment:
sorry, i have one more nitpick.

could you reword this comment slightly? i was confused for a moment why this would cause an invalid handle error; upon further reading i've realized that you meant _this function_ would return said error because `BodyVariant::into_body` will return `None` in the streaming case